### ISSUE TITLE
Same model name, different providers picks the first one always!!!

### ISSUE THREAD

**Issue by nandan121** [coleam00/bolt.new-any-llm#184]
### Describe the bug

I wanted to use GITHUB, (https://github.com/marketplace/models/azure-openai/gpt-4o/playground)

I set the OPENAI_LIKE URL and KEY. Unfortunately, the way it works, it uses the same "name" for model. 

In stream-texts.ts
``const provider = MODEL_LIST.find((model) => model.name === currentModel)?.provider || DEFAULT_PROVIDER;
``
so, it always picks up the first one. I had to put 
```
  { name: 'gpt-4o', label: 'github/gpt-40', provider: 'OpenAILike' },
  { name: 'gpt-4o-mini', label: 'github/gpt-40-mini', provider: 'OpenAILike' },
```
  
  on the top in constants.ts to get it use the GITHUB provider. 

### Link to the Bolt URL that caused the error

usual link

### Steps to reproduce

Add a model with the same name and different provider; and try to get model for both providers work as expected.

### Expected behavior

As before

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 91.1]

**Comment by TableflipFoundry** [comment]
This also happened to me. Burned up all my anthropic tokens and I thought I was using Mixtral all day.

**Comment by TommyHolmberg** [comment]
I found this issue too when trying to make it work with litellm. I have made a pull request with a fix for this particular issue in PR #188 . It still won't work with litellm but I think it's another issue.

**Comment by nandan121** [comment]
> I found this issue too when trying to make it work with litellm. I have made a pull request with a fix for this particular issue in PR #188 . It still won't work with litellm but I think it's another issue.

Many many thanks. I have copied your changes to my branch. Appreciate that you made the changes.

**Comment by syndicate604** [comment]
What I did as it was doing this on Anthropic also 
- remove the default LLM - so it doesnt auto load anything
- always swap to a new provider and then back to swap 

Select OpenAI make a simple call like HI to gpt4
Select Anthropic and change to Haiku 

You cannot just swap from Haiku to Sonnet with the dropdown you need to leave Anthropic and swap to another model etc
